main.nf

```diff
@@ -9,269 +9,1273 @@
+
+//Add decoys if params.add_decoys is set appropriately
+process generate_decoy_database {
+
+    label 'process_very_low'
+    label 'process_single_thread'
+
+    publishDir "${params.outdir}/logs", mode: 'copy', pattern: '*.log'
+
+    input:
+     file(mydatabase) from ch_db_for_decoy_creation
+
+    output:
+     file "${mydatabase.baseName}_decoy.fasta" into searchengine_in_db_decoy_msgf, searchengine_in_db_decoy_comet, pepidx_in_db_decoy, plfq_in_db_decoy
+     file "*.log"
+
+    when:
+     params.add_decoys
+
+    script:
+     """
+     DecoyDatabase  -in ${mydatabase} \\
+                 -out ${mydatabase.baseName}_decoy.fasta \\
+                 -decoy_string ${params.decoy_affix} \\
+                 -decoy_string_position ${params.affix_type} \\
+                 > ${mydatabase.baseName}_decoy_database.log
+     """
+}
+
+// Doesnt work yet. Maybe point the script to the workspace?
+// All the files should be there after collecting.
+//process generate_simple_exp_design_file {
+//    publishDir "${params.outdir}", mode: 'copy'
+//    input:
+//      val mymzmls from mzmls.collect()
+
+//    output:
+//        file "expdesign.tsv" into expdesign
+//    when:
+//        !params.expdesign
+
+//    script:
+//     strng = new File(mymzmls[0].toString()).getParentFile()
+//     """
+//       create_trivial_design.py ${strng} 1 > expdesign.tsv
+//     """
+//}
+
+process openms_peakpicker {
+
+    label 'process_low'
+
+    publishDir "${params.outdir}/logs", mode: 'copy', pattern: '*.log'
+
+    input:
+     tuple mzml_id, path(mzml_file) from mzmls_pp
+
+    when:
+      params.openms_peakpicking
+
+    output:
+     tuple mzml_id, file("out/${mzml_file.baseName}.mzML") into mzmls_comet_picked, mzmls_msgf_picked, mzmls_plfq_picked
+     file "*.log"
+
+    script:
+     in_mem = params.peakpicking_inmemory ? "inmemory" : "lowmemory"
+     lvls = params.peakpicking_ms_levels ? "-algorithm:ms_levels ${params.peakpicking_ms_levels}" : ""
+     """
+     mkdir out
+     PeakPickerHiRes -in ${mzml_file} \\
+                     -out out/${mzml_file.baseName}.mzML \\
+                     -threads ${task.cpus} \\
+                     -debug ${params.pp_debug} \\
+                     -processOption ${in_mem} \\
+                     ${lvls} \\
+                     > ${mzml_file.baseName}_pp.log
+     """
+}
+
+if (params.enzyme == "unspecific cleavage")
+{
+  params.num_enzyme_termini == "none"
+}
+
+pepidx_num_enzyme_termini = params.num_enzyme_termini
+if (params.num_enzyme_termini == "fully")
+{
+  pepidx_num_enzyme_termini = "full"
+}
+
+process search_engine_msgf {
+
+    label 'process_medium'
+
+    publishDir "${params.outdir}/logs", mode: 'copy', pattern: '*.log'
+
+    // ---------------------------------------------------------------------------------------------------------------------
+    // ------------- WARNING: THIS IS A HACK. IT JUST DOES NOT WORK IF THIS PROCESS IS RETRIED -----------------------------
```

Comment:
Should this be documented as a 'known issue' somewhere?

Reply:
Hmm yes I would need to check if this is still the case or if it was fixed. IMHO it is a nextflow problem, I just couldn't nicely describe it yet for a nice issue there.

Reply:
https://github.com/nextflow-io/nextflow/issues/1457

Reply:
That's sad. 
But I guess a warning should be put somewhere that this may occur.

Reply:
I am currently putting it in the Changelog as Known Issues.